pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -48,7 +52,7 @@ in stdenv.mkDerivation {
 
     function fixRunPath {
       p=$(patchelf --print-rpath $1)
-      patchelf --set-rpath "''${p:+$p:}${lib.makeLibraryPath [ stdenv.cc.cc cudatoolkit.lib ]}:${cudatoolkit}/lib:\$ORIGIN/" $1
+      patchelf --set-rpath "''${p:+$p:}${lib.makeLibraryPath [ cc.cc libcublas zlib ]}:\$ORIGIN/" $1
```

Comment:
I guess the central question in terms of whether we can use autopatchelfhook is whether any of these dependencies do not show up in ldd. Otherwise I imagine that it should work.

Reply:
And actually even if there are dependencies that do not show up in ldd, we can always use `addOpenGLRunpath` together with some custom patchelf'ing at the end. So it should def be possible.

Reply:
All of these things are part of DT_NEEDED, all of these things are found through Runpath.
The only thing that isn't part of Runpath by the time I run ldd is libcuda, which isn't part of NEEDED, but must be dlopen()ed at runtime, I guess.

Reply:
I'm not familiar with DT_NEEDED but `libcuda.so.1` has historically been the Achilles heel of autoPatchelfHook. It's my hope that https://github.com/NixOS/nixpkgs/pull/167397 mitigates this pain somewhat.

Reply:
> DT_NEEDED

ELF allows to announce in advance which shared libraries will need to be resolved and loaded at runtime, you can read it e.g. with `patchelf --print-needed lib/libcudnn_cnn_infer.so`. What I wanted to achieve with `find ... ldd | grep` and what I'm now achieving with autoPatchelf is that at least the dependencies declared in this way must be discoverable through runpath. Of course, libcuda has to be handled separately, we just... keep her in mind :(

Reply:
Accidentally, autoPatchelf removes from Rpath things that are not NEEDED -> the problems you mention

Reply:
ah ok so DT_NEEDED is the section that tools like `ldd` inspect to tell us the dependencies?

Reply:
Yes!

Reply:
Using autopatchelfhook now, and in a cleaner fashion